test/test_smartfile.py

```diff
@@ -0,0 +1,50 @@
+import os
+import unittest
+
+from smartfile import BasicClient
+
+API_KEY = os.environ.get("API_KEY")
+API_PASSWORD = os.environ.get("API_PASSWORD")
+
+if API_KEY is None:
+    raise RuntimeError("API_KEY is required")
+
+if API_PASSWORD is None:
+    raise RuntimeError("API_PASSWORD is required")
+
+
+class CustomOperationsTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.current_dir = os.path.dirname(os.path.realpath(__file__)) + "/"
+        self.api = BasicClient(API_KEY, API_PASSWORD)
+        self.txtfile = self.current_dir + "myfile.txt"
+        self.uploaddata = None
+
+    def get_data(self):
+        self.uploaddata = self.api.get("/path/info/myfile.txt")
+        return self.uploaddata
+
+    def upload(self):
+        data = open(self.txtfile, "rb")
+        self.api.upload('myfile.txt', data)
+        self.assertEquals(self.get_data()['size'],
+                          os.path.getsize(self.txtfile))
+
+    def download(self):
+        self.api.download("myfile.txt")
+        f = open('myfile.txt', 'rb')
+        self.assertEquals(f.readlines(), open(self.txtfile, "rb").readlines())
```

Comment:
Instead of checking the files contents, perhaps you might check the file sizes instead. If myfile.txt ends up being big then this assert would take a while.